Feature: Add Post Battle Screen

rankoliang opened the pull request:
This commit
- Counts the dinosaurs when the battle button is pressed
- Adds a counter during the counting phase
- Adds animations when counting the dinosaurs
- Uses the Results Screen when a winner is determined

Resolves #21

rankoliang committed:
Add ScoreResult component

rankoliang committed:
Update transitions to translations

rankoliang committed:
Update vertical divider

rankoliang committed:
Add transition for MainActionButton

rankoliang committed:
Merge main with feature/add-post-battle-screen

rankoliang committed:
Count and animate counting of dinosaurs

rankoliang committed:
Translate dinosaurs vertically when counted

rankoliang committed:
Refactor Level index

rankoliang committed:
Refactor level components

rankoliang committed:
Animate point changes

rankoliang committed:
Add colors to point animations

rankoliang committed:
Animate ScoreResult counter

rankoliang committed:
Refactor counting and transitioning to phases

rankoliang committed:
Add results phase

rankoliang committed:
Fix bug where game would count twice

rankoliang committed:
Update translations when counting

rankoliang committed:
Refactor battlefield

rankoliang committed:
Add key when mapping over dinos

rankoliang committed:
Move data folder

rankoliang committed:
Update styles for results

rankoliang committed:
Add slide in animation to results

rankoliang committed:
Refactor next level

rankoliang committed:
Remount Level component when changing routes

rankoliang committed:
Merge refactor/results with post-battle-screen

rankoliang committed:
Add results menu after dinos are counted

rankoliang committed:
Update min and max counts

rankoliang committed:
Add links and buttons to level

rankoliang committed:
Update rules shown on results screen

rankoliang committed:
Remove unused imports

rankoliang committed:
Move data folder

timjacksonm commented on src/Router.js:
```diff
@@ -10,9 +10,13 @@ const Router = () => {
+          exact
+          path="/levels/:difficulty/:stage"
+          render={() => {
+            return <Level key={Date.now()} />;
```
Glad you found a work around to fix this component re-rendering! Nice work.

timjacksonm commented on src/components/Level/components/Dino.js:
```diff
@@ -0,0 +1,35 @@
+
+const Dino = ({
+  dino: { Component, animating, points, scaleX },
+  dino,
```
A little confusing since dino is repeated twice for the props. You mentioned you might be able to spread that out in the body. If that is possible and looks better lets make that change. Otherwise it works!

timjacksonm commented on src/components/ScoreResult/index.js:
```diff
@@ -0,0 +1,54 @@
+      previousActualScore !== actualScore
+    ) {
+      setTransitioning(true);
+    }
```
Next three comments on this review is something I think would add an indicator on which answer the user got correct after the count is over. Turns the background Green from #333.
If that background is to much we could change the text instead. What do you think?
Add: 
if (score === actualScore && !transitioning) {
      setAnswerColor('Green');
    }

timjacksonm replied:
The way I am suggesting this does have a bug - it changes the background while the score is still being counted. Will update with workaround when I find one.

timjacksonm replied:
Work-around if you add in the phase state as a prop. It doesn't change the background until the phase is set to 'results'
if (phase === 'results' && score === previousActualScore) {
      setAnswerColor('Green');
    }

rankoliang replied:
I added in an animating box-shadow. Let me know if it works!

timjacksonm commented on src/components/ScoreResult/index.js:
```diff
@@ -0,0 +1,54 @@
+}) => {
+  const [transitioning, setTransitioning] = useState(false);
+  const previousActualScore = usePrevious(actualScore);
+
```
Add: 
const [answerColor, setAnswerColor] = useState('#333');

timjacksonm commented on src/components/ScoreResult/index.js:
```diff
@@ -0,0 +1,54 @@
+
+  return (
+    <StyledScoreResult reversed={reversed}>
+      <ScoreResultGroup>
```
Add: 
ScoreResult Group color={answerColor}

timjacksonm commented on src/helpers.js:
```diff
@@ -0,0 +1,32 @@
+import { useRef, useEffect, useState } from 'react';
```
What do you think about renaming this file GlobalHelpers.js?
If you like it don't forget to change it in the files this is used.

timjacksonm reviewed (changes requested):
Thank you for working so hard on this! Just a few changes/suggestions. Great work!

rankoliang committed:
Add glow animation when count is correct

rankoliang committed:
Refactor Dino component props

rankoliang committed:
Move src/helpers to src/globalHelpers

rankoliang committed:
Add level indicator

rankoliang committed:
Fix fossil directions

rankoliang committed:
Update MainActionButton animations

timjacksonm commented on src/components/Level/components/MainActionButton.js:
```diff
@@ -0,0 +1,44 @@
+      color={color}
+      phase={phase}
+      onClick={onClick}
+      phase={phase}
```
phase={phase} listed twice.

timjacksonm reviewed (approved):
Looks good. Just the one eslint error popped up about phase being listed twice.

savwiley committed:
Feature: Changed Index Elements (#29)

* removed old favicon

* added new favicon, changed title & descriptor

* added favicon for other devices

* added android icons, fixed urls

timjacksonm commented on src/components/ScoreResult/styled.js:
```diff
@@ -0,0 +1,80 @@
+import styled, { css, keyframes } from 'styled-components';
+
+const successGlow = keyframes`
```
Personally I kind of like the look of limegreen which is #32CD32 for the success glow.
It might be to much. What do you think?

rankoliang replied:
Looks a lot better!

timjacksonm reviewed (changes requested):
added a new comment.
I did merge the PR for the favicon icon savwiley did. So before we merge I believe we will need to do an update of merging main.

timjacksonm committed:
Update/rules (#30)

* Added blue team dino svgs to legend.

* created level 1 object data, added a new dino type with value of three for each team, adjusted legend to receive level data and display appropriate dinos and their values.

* level updates
-Moved level data object into its own file.
-Added levelData state which determines the level that should be played based on url.
-Adjusted redDinos/blueDinos variables to include levelData. dinoType is the svg image, qty for quantity of that dinoType to put on the field. Removed styling and made a top level variable placeRandom
-Adjusted the jsx to map the dinos to field using levelData.

* refactored the level data based on team feedback.

* updated level data.
-created first 10 levels for beginner difficulty. first 3 levels the winning side is pre determined. After lvl 3 it is random and the legend board is always different from the previous level.

* made more random again

* adjusted naming of props passed to legend.

* destructured set to [dinoId, points] to better explain what set is

* changed teamId to legend.

* updated to be 1 intead of :1. also updated index.js for levels to reflect changes

* changed levels[levelId] to levels[difficulty] and removed previous commit changes.

* refactored dino arrays. installed shuffle-array

* changed require to import statement for shuffle.

* changed getRange function to use reduce method.

* changed 5-point-blue.svg to a different dino svg file. Renamed all svgs to correspond with their id in dinos. Re-wrote imports.

* folder structure
-removed staticLevel folder and added data too src/data.
-moved levels, legends to their own js files inside of src/data.
-renamed index.js to dinos.js and left the dinos variable in the file.
-adjusted imports on level index.js and legend index.js

* updated router for difficulty/stage. enabled prettier with double quote jsx. single quote everything else.

* data updates
-added legend boards that include 4 dinos and also 3 dinos & operand. Have it as subtraction for now.
-added dino11-dino15 for dead dinos to be used as operands.
-created 10 levels for average and hard diffuclty. hard lvl 10 is kind of crazy. It is possible to have a negative value I think so probably won't work. We can change this as we get the rest of the game developed.

* data more random

* deleted dino svgs since they didn't match upstream.

* updated rules

* updated grammar mistakes.

rankoliang committed:
Change highlight glow colors

rankoliang committed:
Update MainActionButton styles

rankoliang committed:
Merge branch 'main' into feature/add-post-battle-screen
